frontend/src/components/ExplorerCard/utils.ts

```diff
@@ -0,0 +1,177 @@
+import axios from 'axios';
+import { SOMETHING_WENT_WRONG } from 'constants/api';
+import { initialQueriesMap } from 'constants/queryBuilder';
+import {
+	queryParamNamesMap,
+	querySearchParams,
+} from 'constants/queryBuilderQueryNames';
+import { mapQueryDataFromApi } from 'lib/newQueryBuilder/queryBuilderMappers/mapQueryDataFromApi';
+import { isEqual } from 'lodash-es';
+
+import {
+	DeleteViewHandlerProps,
+	GetViewDetailsUsingViewKey,
+	IsQueryUpdatedInViewProps,
+	SaveViewHandlerProps,
+	UpdateQueryHandlerProps,
+} from './types';
+
+export const getViewDetailsUsingViewKey: GetViewDetailsUsingViewKey = (
+	viewKey,
+	data,
+) => {
+	const selectedView = data?.find((view) => view.uuid === viewKey);
+	if (selectedView) {
+		const { compositeQuery, name, uuid } = selectedView;
+		const query = mapQueryDataFromApi(compositeQuery);
+		return { query, name, uuid };
+	}
+	return undefined;
+};
+
+export const updateQueryHandler = async ({
+	updateViewAsync,
+	compositeQuery,
+	viewName,
+	viewKey,
+	sourcePage,
+	extraData,
+	setIsQueryUpdated,
+	notifications,
+}: UpdateQueryHandlerProps): Promise<void> => {
+	try {
+		await updateViewAsync({
```

Comment:
move this logic to onSuccess no need for extra handling

Reply:
Sure, removed it.